ansible_base/lib/redis/client.py

```
@@ -20,11 +20,46 @@
 
 _INVALID_STANDALONE_OPTIONS = ['cluster_error_retry_attempts']
 
+# This will eventually be in the Redis package under redis.typing later
+ResponseT = Union[Awaitable, Any]
+
+
+class DABRedisRespectACLFlushMixin:
+    """
+    FLUSHDB is called when we call django.cache.clear()
+
+    FLUSHDB does NOT respect Redis ACL on keys. It flushes all keys in a database.
+    You think your keys are protected because the user you are running FLUSHDB as
+    has write-only access to keys with the pattern gateway* or even read-only access
+    ... WRONG, you aren't safe!
+
+    And if you are thinking of using Redis' command re-write feature to sweep this
+    problem under the rug then god help the developer or support person that comes after you.
+
+    At some point in the future we may also want to override flushall as well
+    """
+
+    def flushdb(self, asynchronous: Optional[bool] = None, **kwargs) -> ResponseT:
+        if asynchronous is not None:
+            logger.warning("DABRedis clients implement an ACL friendly FLUSHDB which can not be async")
```

Just reading over the tests and the question popped into my head. Would it be better to raise an exception here, `ValueError`, to notify the developer that `asynchronous` isn't supported?

---

I was concerned that if there was another library that was passing us the async it might be out of the devs control. i.e. if django Redis cache changed to pass async=True there is nothing a dev could really do about it.